virttest/vt_utilities/cpu.py

```diff
@@ -0,0 +1,173 @@
+                                   "avx512vbmi,avx512_vbmi2|avx512vbmi2,"
+                                   "hle,rtm"
+                                   "gfni,vaes,vpclmulqdq,avx512_vnni,hle,rtm"),
+                "Icelake-Client-noTSX": ("avx512_vpopcntdq|avx512-vpopcntdq,"
+                                         "avx512vbmi,avx512_vbmi2|avx512vbmi2,"
+                                         "gfni,vaes,vpclmulqdq,avx512_vnni"),
+                "Cascadelake-Server": ("avx512f,avx512dq,avx512bw,avx512cd,"
+                                       "avx512vl,clflushopt,avx512_vnni,hle,"
+                                       "rtm"),
+                "Cascadelake-Server-noTSX": ("avx512f,avx512dq,avx512bw,"
+                                             "avx512cd,avx512vl,clflushopt,"
+                                             "avx512_vnni"),
+                "Skylake-Server": "avx512f,clwb,xgetbv1,pcid,hle,rtm",
+                "Skylake-Server-noTSX-IBRS": "avx512f,clwb,xgetbv1,pcid",
+                "Skylake-Client": "xgetbv1,pcid,hle,rtm",
+                "Skylake-Client-noTSX-IBRS": "xgetbv1,pcid",
+                "Broadwell": "adx,rdseed,3dnowprefetch,hle,rtm",
+                "Broadwell-noTSX": "adx,rdseed,3dnowprefetch",
+                "Haswell": "fma,avx2,movbe,hle,rtm",
+                "Haswell-noTSX": "fma,avx2,movbe",
+                "IvyBridge": "f16c,fsgsbase,erms",
+                "SandyBridge": ("avx,xsave,aes,sse4_2|sse4.2,sse4.1|sse4_1,"
+                                "cx16,ssse3"),
+                "Westmere": "aes,sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                "Nehalem": "sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                "Penryn": "sse4.1|sse4_1,cx16,ssse3",
+                "Conroe": "ssse3"}
+
+
+def get_cpu_info():
+    """
+    Return information about the CPU architecture
+
+    :return: cpu information in dict type
+    """
+    cmd = "lscpu | tee"
+    output = process.run(cmd, shell=True,
+                         ignore_status=True).stdout_text.splitlines()
+    return dict(map(lambda x: [i.strip() for i in x.split(":", 1)], output))
+
+
+def get_host_cpu_models():
+    """
+    Get cpu model from host cpuinfo
+    """
+    def _cpu_flags_sort(cpu_flags):
+        """
+        Update the cpu flags get from host to a certain order and format
+        """
+        flag_list = sorted(re.split("\s+", cpu_flags.strip()))
+        cpu_flags = " ".join(flag_list)
+        return cpu_flags
+
+    def _make_up_pattern(flags):
+        """
+        Update the check pattern to a certain order and format
+        """
+        pattern_list = sorted(re.split(",", flags.strip()))
+        pattern = r"(\b%s\b)" % pattern_list[0]
+        for i in pattern_list[1:]:
+            pattern += r".+(\b%s\b)" % i
+        return pattern
+
+    if ARCH in ('ppc64', 'ppc64le'):
+        return []  # remove -cpu and leave it on qemu to decide
```

Comment:
We should update this part to adopt the current requirements of designing we wanted. This part of handling should be not decided by the virtualization testing layer.